pkg/tunnel/proxy/common/proxy.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2020 The SuperEdge Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"bytes"
+	uuid "github.com/satori/go.uuid"
+	"github.com/superedge/superedge/pkg/tunnel/conf"
+	"github.com/superedge/superedge/pkg/tunnel/context"
+	"github.com/superedge/superedge/pkg/tunnel/proxy/modules/stream/streammng/connect"
+	"github.com/superedge/superedge/pkg/tunnel/util"
+	"io"
+	"k8s.io/klog/v2"
+	"net"
+	"strings"
+)
+
+func ProxyEdgeNode(nodename, host, port, category string, proxyConn net.Conn, req *bytes.Buffer) {
+	node := context.GetContext().GetNode(nodename)
+	if node != nil {
+		//If the edge node establishes a long connection with this pod, it will be forwarded directly
+		uid := uuid.NewV4().String()
+		ch := context.GetContext().AddConn(uid)
+		node.BindNode(uid)
+		_, err := proxyConn.Write([]byte(util.ConnectMsg))
+		if err != nil {
+			klog.Errorf("Failed to write data to proxyConn, error: %v", err)
+			return
+		}
+		go Read(proxyConn, node, category, util.TCP_FRONTEND, uid, host+":"+port)
+		Write(proxyConn, ch)
+	} else {
+		//From tunnel-coredns, query the pods of tunnel-cloud where edge nodes establish long-term connections
+		var remoteConn net.Conn
+		addrs, err := net.LookupHost(nodename)
+		if err != nil {
+			if dnsErr, ok := err.(*net.DNSError); ok {
+				if dnsErr.IsNotFound {
+					remoteConn, err = net.Dial("tcp", host+":"+port)
```

Comment:
should we cache this proxyconn in context.GetContext()?  Or we maybe need close both remoteconn proxyconn